erpnext/accounts/doctype/budget/budget.py

```diff
@@ -14,20 +14,24 @@ class DuplicateBudgetError(frappe.ValidationError): pass
 
 class Budget(Document):
 	def autoname(self):
-		self.name = make_autoname(self.cost_center + "/" + self.fiscal_year + "/.###")
+		self.name = make_autoname(self.get(frappe.scrub(self.budget_against)) + "/" + self.fiscal_year + "/.###")
 
```

Comment:
Actually there is  -  already in cost center and fiscal year naming so using that again make it very hard to understand the name.
Ex.  Mumbai - MPC/2016-2017/001